clients/src/test/java/org/apache/kafka/clients/admin/MockAdminClient.java

```diff
@@ -38,13 +38,25 @@
 import java.util.concurrent.TimeUnit;
 
 public class MockAdminClient extends AdminClient {
+    public static final String CLUSTER_ID = "I4ZmrWqfT2e-upky_4fdPA";
+
     private final List<Node> brokers;
     private final Map<String, TopicMetadata> allTopics = new HashMap<>();
 
+    private Node controller;
     private int timeoutNextRequests = 0;
 
+    /**
+     * Creates MockAdminClient for a cluster with the given brokers. By default the first broker in the list is the controller
+     * @param brokers list of brokers in the cluster
+     */
     public MockAdminClient(List<Node> brokers) {
         this.brokers = brokers;
+        this.controller = brokers.get(0);
+    }
+
+    public void controller(Node controller) {
```

Comment:
Seems we don't use this anywhere, but I guess there's no harm having it. Not sure it matters, but maybe we should have a sanity check that the controller is included in the broker list?